hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/pages/datanodes/datanodes.tsx

```diff
@@ -0,0 +1,312 @@
+    setSelectedRows([]);
+  };
+
+  const { dataSource, lastUpdated, columnOptions } = state;
+
+  return (
+    <>
+      <div className='page-header-v2'>
+        Datanodes
+        <AutoReloadPanel
+          isLoading={loading}
+          lastRefreshed={lastUpdated}
+          togglePolling={autoReloadHelper.handleAutoReloadToggle}
+          onReload={loadData} />
+      </div>
+      <div style={{ padding: '24px' }}>
+        <div className='content-div'>
+          <div className='table-header-section'>
+            <div className='table-filter-section'>
+              <MultiSelect
+                options={columnOptions}
+                defaultValue={selectedColumns}
+                selected={selectedColumns}
+                placeholder='Columns'
+                onChange={handleColumnChange}
+                onTagClose={() => { }}
+                fixedColumn='hostname'
+                columnLength={columnOptions.length} />
+              {selectedRows.length > 0 &&
+                <Button
+                  type="primary"
+                  icon={<DeleteOutlined />}
+                  style={{
+                    background: '#FF4D4E',
+                    borderColor: '#FF4D4E'
+                  }}
+                  loading={loading}
+                  onClick={() => { setModalOpen(true) }}> Remove
+                </Button>
+              }
+            </div>
+            <Search
+              disabled={dataSource?.length < 1}
+              searchOptions={SearchableColumnOpts}
+              searchInput={searchTerm}
+              searchColumn={searchColumn}
+              onSearchChange={
+                (e: React.ChangeEvent<HTMLInputElement>) => setSearchTerm(e.target.value)
+              }
+              onChange={(value) => {
+                setSearchTerm('');
+                setSearchColumn(value as 'hostname' | 'uuid' | 'version' | 'revision')
+              }} />
+          </div>
+          <DatanodesTable
+            loading={loading}
+            data={dataSource}
+            selectedColumns={selectedColumns}
+            selectedRows={selectedRows}
+            searchColumn={searchColumn}
+            searchTerm={debouncedSearch}
+            handleSelectionChange={handleSelectionChange}
+            decommissionUuids={decommissionUuids} />
+        </div>
+      </div>
+      <Modal
+          title=''
+          centered={true}
+          visible={modalOpen}
+          onOk={handleModalOk}
+          onCancel={handleModalCancel}
+          closable={false}
+          width={400} >
+            <div style={{
+              margin: '0px 0px 5px 0px',
+              fontSize: '16px',
+              fontWeight: 'bold'
+            }}>
+              <WarningFilled className='icon-warning' style={{paddingRight: '8px'}}/>
+              Stop Tracking Datanode
+            </div>
+            Are you sure, you want recon to stop tracking the selected <strong>{selectedRows.length}</strong> datanode(s)?
```

Comment:
Another issue I encountered is that when all datanodes are removed from the list (e.g., by shutting them down), the new UI crashes and displays an error. This behavior wasn’t present in the old UI, which handled this case by showing an empty list.

<img width="944" alt="image" src="https://github.com/user-attachments/assets/36d59162-b4a1-4cd5-97fe-6a8c68a843dc">

Reply:
Is this issue taken care ?

Reply:
Yes @devmadhuu, this has been fixed and tested out after building the new changes as well